### PR TITLE
Cache removal method

### PR DESCRIPTION
**Pull request by toohotz**
Given that we have methods already that adds values into the cache, figured I'd poke at removing values out of the cache.  This PR will deal with removing a single value out of a cache as well as stopping any observers from observing this value once it has been successfully removed.  This will also introduce the `FlatCacheError` that will handle any errors that can occur.

There will tests to verify these proposed changes.

Always open for suggestions on how to make this implementation better as its my first stab at it.

### CODE REVIEW

File: FlatCache/FlatCache.swift [hunk 1]
```diff
@@ -32,7 +32,7 @@ private extension Cachable {
     static var typeName: String {
         return String(describing: self)
     }
-
+
```
Comment (rnystrom) [inline FlatCache/FlatCache.swift]: Can you strip these whitespace changes?

File: FlatCache/FlatCache.swift [hunk 2]
```diff
@@ -128,13 +129,33 @@ public final class FlatCache {
...
+        }
+        storage.removeValue(forKey: key)
+        enumerateListeners(key: key) { listener in
+            listener.flatCacheDidUpdate(cache: self, update: FlatCache.Update.removeItem(val))
+            if let index = listeners.index(forKey: key) {
+                listeners.remove(at: index)
+            }
+        }
+        return val as? T
+    }
```
Comment (rnystrom) [inline FlatCache/FlatCache.swift:160]: What information does the `throw` provide that a `nil` return value doesn't?

File: FlatCacheTests/FlatCacheTests.swift [hunk 3]
```diff
@@ -118,4 +122,44 @@ class FlatCacheTests: XCTestCase {
...
+        do {
+            let cachedVal = try cache.remove(key: m1.id) as CacheModel?
+            let _ =  try cache.remove(key: m3.id) as CacheModel?
+            XCTAssertEqual(m1.value, cachedVal?.value)
+            XCTAssertEqual(l1.receivedItemQueue.count, 0)
+            XCTAssertEqual(l2.receivedItemQueue.count, 1)
+        } catch {
+            XCTAssert(false, "\(error)")
+        }
+    }
```
Comment (rnystrom) [inline FlatCacheTests/FlatCacheTests.swift:164]: Yaaaay tests!

**Review by rnystrom — CHANGES_REQUESTED** [review]

File: FlatCache/FlatCache.swift (hunk 2 above)
Comment (toohotz) [reply FlatCache/FlatCache.swift:160]: So the information that the error would throw that would be useful is the key that there was no value for.  It then becomes a question of is it useful to do the try catch dance as opposed to just outright passing up the optional value.  What do you think?

File: FlatCache/FlatCache.swift (hunk 1 above)
Comment (toohotz) [reply FlatCache/FlatCache.swift]: Just noticed that on my new lines, Xcode has it setup to auto indent so, oh Xcode.. 🙄

**Pull request closed without merging**